packages/api/src/controllers/DocumentsController.js

```diff
@@ -1,20 +1,54 @@
 const DocumentsDAO = require('../dao/DocumentsDAO')
+const { decodeStateTransition } = require('../utils')
 
 class DocumentsController {
-  constructor (knex) {
+  constructor (client, knex, dapi) {
     this.documentsDAO = new DocumentsDAO(knex)
+    this.client = client
+    this.dapi = dapi
   }
 
   getDocumentByIdentifier = async (request, response) => {
     const { identifier } = request.params
 
-    const document = await this.documentsDAO.getDocumentByIdentifier(identifier)
+    const documentData = await this.documentsDAO.getDocumentData(identifier)
 
-    if (!document) {
+    // second for system documents, like dash.dash alias
+    if (!documentData || !documentData?.transition_data) {
```

Comment:
What is `transition_data`?

Reply:
I thnk that the logic is not all the way correct. We need to query the database first, and if it wasn't found, then search in the DAPI. Or perhaps query DAPI anyway and merge two structures together? Are there any other interesting fields missing in our indexer?